packages/core/strapi/lib/core/loaders/components.js

```diff
@@ -6,19 +6,20 @@ const { pathExists } = require('fs-extra');
 const loadFiles = require('../../load/load-files');
 
 module.exports = async strapi => {
-  if (!(await pathExists(strapi.dirs.components))) {
+  if (!(await pathExists(strapi.dirs.dist.components))) {
     return {};
   }
 
-  const map = await loadFiles(strapi.dirs.components, '*/*.*(js|json)');
+  const map = await loadFiles(strapi.dirs.dist.components, '*/*.*(js|json)');
 
   return Object.keys(map).reduce((acc, category) => {
     Object.keys(map[category]).forEach(key => {
       const schema = map[category][key];
 
-      const filePath = join(strapi.dirs.components, category, schema.__filename__);
-
       if (!schema.collectionName) {
+        // NOTE: We're using the filepath from the app directory instead of the dist for information purpose
+        const filePath = join(strapi.dirs.app.components, category, schema.__filename__);
```

Comment:
The filepath is computed to warn the user about a potential issue in the component schema. Thus, it makes more sense (IMO) to display the path of the source file for this schema (app dir) instead of the built version (dist). What do you think?